k2/csrc/fsa_equivalent.cc

```diff
@@ -62,8 +90,63 @@ bool IsRandEquivalent(const Fsa &a, const Fsa &b, std::size_t npath /*=100*/) {
   return true;
 }
 
-bool RandomPath(const Fsa &a, Fsa *b,
-                std::vector<int32_t> *state_map /*=nullptr*/) {
+template <class FbWeightType>
+bool IsRandEquivalent(const Fsa &a, const float *a_weights, const Fsa &b,
```

Comment:
Should we place it in the header file? 

Since it is a template and it is not instantiated in this `.cc` file, we may have compiler errors
later if it is used in another `.cc` file.

Reply:
Or we can do explicitly instantiation in this `.cc` file and place an `extern template`
declaration in the header file.

Reply:
Yes, I plan do `explicit template instantiation` here as there are only two different types for FbWeightType